enhancements/update/configurable-clusterversion-signature-stores.md

```diff
@@ -0,0 +1,194 @@
+approvers:
+  - "@LalatenduMohanty"
+api-approvers:
+  - "@deads2k"
+creation-date: 2023-09-27
+last-updated: 2023-10-31
+tracking-link:
+  - https://issues.redhat.com/browse/OTA-916
+---
+
+# ClusterVersion has option to replace default signature stores
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] Operational readiness criteria is defined
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+CVO currently fetches the upstream urls for signature verification through [cluster-update-keys]. 
+To increase flexibility and improve disconnected cluster user experience, we need to provide a way for the CVO to 
+get the required signatures from custom signature stores or the OSUS running in the disconnected environment.
+
+
+## Motivation
+
+Today, in the disconnected environments, a user has to clone the signatures config map as well as the release images 
+then apply the configMap to each cluster for the update to proceed successfully.
+With [serve image signatures for disconnected environments], OSUS will be able to serve the cloned image signatures locally.
+
+Currently, there is no way for the CVO to fetch the signatures through custom URIs. This enhancement defines an API change in the CVO
+to accept a custom update-service URI which can be used to verify release signatures. 
+
+
+### User Stories
+
+* As a cluster administrator, I want the cluster's CVO to add new signature store URIs, so that I can have local release signature 
+  verification in air gapped environments.
+
+### Goals
+
+* To verify the release image signature in disconnected environments using OSUS signatures URI.
+* To replace existing signature stores with custom user-provided stores.
+
+### Non-Goals
+
+* This enhancement does not cover how OSUS running in the disconnected environment gets the signatures and if the 
+signatures are validated during the mirroring process.
+
+## Proposal
+
+### Workflow Description
+
+1. Cluster admin will have to add the verification URI to CVO
+2. CVO will reach the newly added URI to verify the release images.
+
+### API Extensions
+
+* CVO will grow a new field, an array, which will be populated with the image verification URI.
+* This will be an array as it'll allow the admin to use multiple locally configured stores. 
+* Existing default signature stores will be used as fallback in-case custom stores don't contain signatures or are offline. 
+* If this property is empty, default stores will be used, similar to how we handle upstream for update service today.
+* Admins will still be able to use the existing config maps for signature verification. 
+* CVO will check the release signatures in the local ConfigMap first and will reach in parallel to other signature stores.
```

Comment:
Does it already do this?

So does this new API become a second way to specify the same information that already exists in the configmap? If the configmap is defined by some reference, we should probably deprecate the reference and promote users to move to the new API

Reply:
yup, we currently have [docs][1] like:

> If you mirrored release images, apply the release image signatures to the cluster by running the following command:
>
> ```console
> $ oc apply -f ./oc-mirror-workspace/results-1639608409/release-signatures/
> ```

That this ability to configure `signatureStores` will allow us to slowly deprecate.  Although we may not remove them completely, because the maintenance load of keeping the local ConfigMap signature store working is not high, and maybe it would help some customers.

[1]: https://docs.openshift.com/container-platform/4.14/updating/updating_a_cluster/updating_disconnected_cluster/mirroring-image-repository.html#oc-mirror-updating-cluster-manifests_mirroring-ocp-image-repository